From 61bee84d8afc42576d309a616c3894f18c1b964f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Daniel=20Vr=C3=A1til?= <dvratil@kde.org>
Date: Wed, 19 Apr 2017 01:03:56 +0200
Subject: [IMAP] Fix IDLE support with GSSAPI authentication

Only ignore IDLE due to empty password when not using GSSAPI.
With GSSAPI the password field in settings is always empty.

BUG: 311456
FIXED-IN: 5.5.1
---
 resources/imap/imapresourcebase.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/resources/imap/imapresourcebase.cpp b/resources/imap/imapresourcebase.cpp
index cd2a04a..10a9bad 100644
--- a/resources/imap/imapresourcebase.cpp
+++ b/resources/imap/imapresourcebase.cpp
@@ -587,7 +587,9 @@ void ImapResourceBase::startIdle()
     }
 
     //Without password we don't even have to try
-    if (settings()->password().isEmpty()) {
+    if (m_pool->account()->authenticationMode() != KIMAP::LoginJob::GSSAPI &&
+        settings()->password().isEmpty())
+    {
         return;
     }
 
-- 
cgit v0.11.2

